Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py

```diff
@@ -111,7 +111,8 @@ def validater(username, access_token):
             expiration = response.headers.get('github-authentication-token-expiration', None)
             if expiration:
                 expiration = int(calendar.timegm(datetime.strptime(expiration, '%Y-%m-%d %H:%M:%S UTC').timetuple()))
-            if (expiration is None or expiration > time.time()) and response.status_code == 200 and response.json().get('login') == username:
+                expiring_soon = 12 * 60 * 60  # Consider a token expiring in the next 12 hours to be expired
```

Comment:
> Consider a token expiring in the next 12 hours to be expired

Why? If it's not expired yet, what's the problem in using that unexpired token?

Instead can we print a warning that token is going to expire in next few days? (maybe start 5 days before)

Reply:
`git-webkit setup` is the command that actually runs this check. We don't do a validity check for every request to GitHub because doing a validity check involve making a request. That means we're only really running this check when folks are setting up their repository, and it seems bad to allow users to set up a repository that will be potentially invalid later that same day.